commands/timeout.js

```diff
@@ -1,21 +1,33 @@
+const Discord = require('discord.js');
 const firebase = require('firebase');
-const moment = require('moment');
-const format = require('../momentFormat');
+const splitargs = require('splitargs');
+const juration = require('juration');
 
 module.exports = {
-  usage: '[@user]',
+  usage: '[@user] [30m]',
   description: 'ADMIN ONLY: Give a user the \'Restricted\' role for 30 minutes. Will also remove the \'18+\' role.',
   allowDM: false,
   requireRoles: ['Admin', 'Moderator'],
   process: (bot, message) => {
-    let timeoutEnd = Date.now() + 1800000;
-
-    const userLogs = bot.channels.find('name', 'user-logs');
+    const userLogsChannel = bot.channels.find('name', 'user-logs');
 
     const restrictedRole = message.guild.roles.find('name', 'Restricted');
     const over18Role = message.guild.roles.find('name', '18+');
     const memberRole = message.guild.roles.find('name', 'Member');
 
+    // Calculate the end time
+    const timeoutStart = Date.now();
+
+    // Let's grab just the time...
+    let timeoutLengthArray = splitargs(message.content);
+    timeoutLengthArray.shift();
+    timeoutLengthArray.shift();
+    let timeoutLength = timeoutLengthArray[0];
+
+    if (!timeoutLength) timeoutLength = '30m';
```

Comment:
Thoughts on explicit blocks?

ie even though this is doesn't need them,
```javascript
if (!timeoutLength) {
  timeoutLength = '30m';
}
```

This seems to be agreed to be worth it in some circles, like how it fucked Apple with their goto fail 😛